### DISCUSSION TITLE
Determine the order of middleware

### DISCUSSION THREAD

**Discussion by agracia-foticos** [nuxt/nuxt#18736] [Questions] [answered]
Hello,

We need to determine the order of middleware.

We have 2 middleware, one creates a cookie with the id of the session, and the another gets this cookie and uses a fetch with the id of session.

How we can determine the order of middleware?? there is anything similar that the order of plugins?

Thanks in advance!

Best regards

**Comment by TheAlexLichter** [comment, marked as answer]
Does numbering them (like for plugins) doesn't work? (Like `1.create-cookie.ts`, `2.fetch-session.ts`?)

**Reply by agracia-foticos** [reply]
> Does numbering them (like for plugins) doesn't work? (Like `1.create-cookie.ts`, `2.fetch-session.ts`?)

seems to work

**Comment by mercs600** [comment]
@manniL what about middleware attached by module? for example:
```js
  addRouteMiddleware('01.module-middleware', moduleMiddleware, {
    global: true
  })
```

Am I able to reorder them?
